### PR TITLE
Fix Python versions

### PR DESCRIPTION
**Pull request by SpacemanPaul**
Updated supported Python version to >= 3.9 throughout.  
Updated test conda environment to Python 3.9
Pinned odc_apps_dc_tools to Python 3.9-3.10 as per https://github.com/opendatacube/datacube-core/issues/1511

### CODE REVIEW

File: tests/test-env-py39.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ dependencies:
   - pytest-httpserver
   - pytest-cov
   - pytest-timeout
-  - moto
+  - moto[server]
```
Comment (Kirill888) [inline tests/test-env-py39.yml]: not valid conda syntax, this only works with `pip`, unless something changed recently in conda

Comment (SpacemanPaul) [reply tests/test-env-py39.yml]: Yeah, I had a feeling that was the case.  I've installed conda and I'll deal with it tomorrow.

Comment (SpacemanPaul) [reply tests/test-env-py39.yml]: It's the moto version that's the issue - conda is pulling in an ancient version for some reason.

Comment (SpacemanPaul) [reply tests/test-env-py39.yml]: Lol.  I just removed the ancient pin on `aiobotocore` and it built in 1/5 the time and works fine.

**Review by emmaai — APPROVED** [review]

File: apps/dc_tools/setup.cfg [hunk 2]
```diff
@@ -16,7 +16,7 @@ url = https://github.com/opendatacube/odc-tools/
 include_package_data = true
 zip_safe = false
 packages = find_namespace:
-python_requires = >=3.7
+python_requires = >= 3.9,<=3.10
```
Comment (omad) [inline apps/dc_tools/setup.cfg]: At least make it up to and include 3.11. I'd prefer not to have an upper bound on Python version.

File: tests/test-env-py39.yml [hunk 3]
```diff
@@ -49,7 +49,6 @@ dependencies:
   - moto
   - deepdiff
 
-
   # for docs
   - sphinx
```
Comment (omad) [inline tests/test-env-py39.yml:53]: Don't do it now, but, omg, testing everywhere is such a mess. I'm sure we don't need sphinx here...

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -28,7 +28,7 @@ jobs:
     - name: Setup Python
       uses: actions/setup-python@v1
       with:
-        python-version: 3.8
+        python-version: 3.9
```
Comment (omad) [inline .github/workflows/main.yml:31]: At some point, we should use the GitHub actions matrix to parameterise this and test more than a single Python version.

**Review by omad — CHANGES_REQUESTED** [review]
Thanks Paul. Other than the <=3.10 restriction on Python this looks okay.

File: apps/dc_tools/setup.cfg (hunk 2 above)
Comment (SpacemanPaul) [reply apps/dc_tools/setup.cfg]: Upper bound was because of issue with aiohttp reported in https://github.com/opendatacube/datacube-core/issues/1511

**Review by omad — APPROVED** [review]
ty

**Pull request merged**